NitroxClient/MonoBehaviours/DiscordRP/DiscordController.cs

```diff
@@ -0,0 +1,196 @@
+        }
+
+        public void DisconnectedCallback(int errorCode, string message)
+        {
+            Log.Info(string.Format("Discord: disconnect {0}: {1}", errorCode, message));
+        }
+
+        public void ErrorCallback(int errorCode, string message)
+        {
+            Log.Error(string.Format("Discord: error {0}: {1}", errorCode, message));
+        }
+
+        public void JoinCallback(string secret)
+        {
+            Log.Info(string.Format("Discord: join ({0})", secret));
+            Log.Debug("Discord" + SceneManager.GetActiveScene().name);
+            if (SceneManager.GetActiveScene().name == "StartScreen")
+            {
+                NitroxServiceLocator.BeginNewLifetimeScope();
+                JoinServer.SaveGameMenuPrototype = FindObject(MainMenuRightSide.main.gameObject, "SavedGames");
+                JoinServer joinServer = gameObject.AddComponent<JoinServer>();
+                string[] serverIpPort = secret.Split(':');
+                joinServer.ServerIp = serverIpPort[0];
+                joinServer.serverPort = int.Parse(serverIpPort[1]);
+            }
+            else
+            {
+                Log.InGame("Please be in the mainmenu if you want to join a session.");
+                Log.Warn("Discord: Can't join a server outside of the mainmenu.");
+            }
+        }
+
+        public void SpectateCallback(string secret)
+        {
+            Log.Info(string.Format("Discord: spectate ({0})", secret));
+        }
+
+        public void RequestCallback(ref DiscordRpc.DiscordUser request)
+        {
+            if (!ShowingWindow)
+            {
+                Log.Info(string.Format("Discord: join request {0}#{1}: {2}", request.username, request.discriminator, request.userId));
+                AcceptRequest acceptRequest = gameObject.AddComponent<AcceptRequest>();
+                acceptRequest.Request = request;
+                lastJoinRequestUserID = request.userId;
+                ShowingWindow = true;
+            }
+            else
+            {
+                Log.Info("Discord: Request window is allready active.");
+            }
+        }
+
+        void Update()
+        {
+            DiscordRpc.RunCallbacks();
+        }
+
+        void OnEnable()
+        {
+            Log.Info("Discord: init");
+            handlers = new DiscordRpc.EventHandlers();
+            handlers.readyCallback = ReadyCallback;
+            handlers.disconnectedCallback += DisconnectedCallback;
+            handlers.errorCallback += ErrorCallback;
+            handlers.joinCallback += JoinCallback;
+            handlers.spectateCallback += SpectateCallback;
+            handlers.requestCallback += RequestCallback;
+            DiscordRpc.Initialize(applicationId, ref handlers, true, optionalSteamId);
+
+            ShowingWindow = false;
+        }
+
+        void OnDisable()
+        {
+            Log.Info("Discord: shutdown");
+            DiscordRpc.Shutdown();
```

Comment:
Unregister the `handlers` callbacks here.

Reply:
I have this code from the official discord example, so I don't think I have to do this because it completely shuts down.

Reply:
I suggest that every fucking one of you leave or just continue this account before the FBI contacts you, because it's only a matter of time before they will. You fucked with the wrong one this time!